### ISSUE TITLE
Data race issues during parallel request execution

### ISSUE THREAD

**Issue by maxkondr** [okta/okta-sdk-golang#257]
Hi, 

We use this lib v2.6.1 for our services. The code is pretty simple and looks like 
```
s.c := okta.NewClient(
		ctx,
		okta.WithOrgUrl(u.String()),
		okta.WithToken(token),
		okta.WithHttpClientPtr(&http.Client{
			Transport: logger.HTTP(http.DefaultTransport, l.Debugf),
		}),
		okta.WithCache(false))
```

```
u := okta.CreateUserRequest{
		Profile: &okta.UserProfile{
			"email":     login,
			"login":     login,
			"firstName": firstName,
			"lastName":  lastName,
		},
}
qp := query.NewQueryParams(query.WithActivate(true))
user, _, err := s.c.User.CreateUser(ctx, u, qp)
```
and 
```
user, _, err := s.c.User.GetUser(ctx, url.QueryEscape(login))
```

and similar.

But once we run tests with `-race` flag we see the following:
```
WARNING: DATA RACE
Write at 0x00c000286150 by goroutine 76:
  github.com/okta/okta-sdk-golang/v2/okta.(*RequestExecutor).WithAccept()
      /Users/yakut/go_path/pkg/mod/github.com/okta/okta-sdk-golang/v2@v2.6.1/okta/requestExecutor.go:236 +0x135
  github.com/okta/okta-sdk-golang/v2/okta.(*UserResource).GetUser()
      /Users/yakut/go_path/pkg/mod/github.com/okta/okta-sdk-golang/v2@v2.6.1/okta/user.go:74 +0xc5
  github.com/<...>/okta.(*provider).FindUser()
      /Users/yakut/go_path/src/github.com/<...>/okta/okta.go:127 +0x110
  github.com/<...>/okta.TestFindUser.func3()
      /Users/yakut/go_path/src/github.com/<...>/okta/okta_test.go:306 +0xbd
  testing.tRunner()
      /usr/local/Cellar/go/1.16.3/libexec/src/testing/testing.go:1193 +0x202

Previous write at 0x00c000286150 by goroutine 75:
  github.com/okta/okta-sdk-golang/v2/okta.(*RequestExecutor).WithAccept()
      /Users/yakut/go_path/pkg/mod/github.com/okta/okta-sdk-golang/v2@v2.6.1/okta/requestExecutor.go:236 +0x135
  github.com/okta/okta-sdk-golang/v2/okta.(*UserResource).GetUser()
      /Users/yakut/go_path/pkg/mod/github.com/okta/okta-sdk-golang/v2@v2.6.1/okta/user.go:74 +0xc5
  github.com/<...>/okta.(*provider).FindUser()
      /Users/yakut/go_path/src/github.com/<...>/okta/okta.go:127 +0x110
  github.com/<...>/okta.TestFindUser.func2()
      /Users/yakut/go_path/src/github.com/<...>/okta/okta_test.go:298 +0xaf
  testing.tRunner()
      /usr/local/Cellar/go/1.16.3/libexec/src/testing/testing.go:1193 +0x202

Goroutine 76 (running) created at:
  testing.(*T).Run()
      /usr/local/Cellar/go/1.16.3/libexec/src/testing/testing.go:1238 +0x5d7
  github.com/<...>/okta.TestFindUser()
      /Users/yakut/go_path/src/github.com/<...>/okta/okta_test.go:303 +0x35c
  testing.tRunner()
      /usr/local/Cellar/go/1.16.3/libexec/src/testing/testing.go:1193 +0x202

Goroutine 75 (running) created at:
  testing.(*T).Run()
      /usr/local/Cellar/go/1.16.3/libexec/src/testing/testing.go:1238 +0x5d7
  github.com/<...>/okta.TestFindUser()
      /Users/yakut/go_path/src/github.com/<...>/okta/okta_test.go:295 +0x299
  testing.tRunner()
      /usr/local/Cellar/go/1.16.3/libexec/src/testing/testing.go:1193 +0x202
```

To dig deeper we see that in the lib the following code is executed
```
func (m *UserResource) CreateUser(ctx context.Context, body CreateUserRequest, qp *query.Params) (*User, *Response, error) {
	...
	req, err := m.client.requestExecutor.WithAccept("application/json").WithContentType("application/json").NewRequest("POST", url, body)
}
```

the issue is in `WithAccept` and `WithContentType` calls as soon as they change the same field in global `RequestExecutor` like:

```
func (re *RequestExecutor) WithAccept(acceptHeader string) *RequestExecutor {
	re.headerAccept = acceptHeader
	return re
}

func (re *RequestExecutor) WithContentType(contentTypeHeader string) *RequestExecutor {
	re.headerContentType = contentTypeHeader
	return re
}
```

Please advise how to overcome this issue. Setting a mutex (or any another sync primitives) around our code is the last choice for us because it decreases performance drastically.

**Comment by bogdanprodan-okta** [comment]
Hi @maxkondr! Thanks for submitting this issue! This seems like a major flaw so we will try to solve this issue asap.

**Comment by monde** [comment]
fixed #256 , released as 2.6.2, thanks @maxkondr , @bogdanprodan-okta
